onnxruntime/core/optimizer/common_subexpression_elimination.cc

```diff
@@ -318,7 +318,13 @@ struct NodeArgPtrEquality {
 };
 
 bool IsNodeSupported(const Node& node) {
-  return !node.ContainsSubgraph() && optimizer_utils::IsOperationDeterministic(node.Domain(), node.OpType());
+  // skip control flow nodes, nodes that produce non-deterministic output, and DequantizeLinear (DQ) nodes.
+  // the reason for skipping DQ is that the QDQ handling looks for QDQ node groups (DQ -> fp32 node -> Q node)
+  // and does not allow for a DQ node to be used in multiple groups. coalescing multiple DQ nodes into one
+  // would result in it having multiple consumers for its output, and it being used in multiple QDQ node groups.
+  return !node.ContainsSubgraph() &&
+         optimizer_utils::IsOperationDeterministic(node.Domain(), node.OpType()) &&
+         !(node.Domain() == kOnnxDomain && node.OpType() == "DequantizeLinear");
```

Comment:
do we also need to check that domain is kOnnxDomainAlias? like here:
https://github.com/microsoft/onnxruntime/blob/6f844522c8c5c2b1125dbfe9de99e1ddfb21a470/onnxruntime/core/graph/graph_utils.cc#L222

Reply:
I don't think so as Node::Init has this:

```c++
  if (kOnnxDomainAlias == domain_) {
    domain_ = kOnnxDomain;
  }
```

Based on that I don't think we would see a node that is loaded from a model with a domain of kOnnxDomainAlias. 

I couldn't find any places where a node was dynamically created that used kOnnxDomainAlias either.

I can cleanup the existing unnecessary checks for kOnnxDomainAlias that create confusion in a separate PR.  
Seems like we only need to check kOnnxDomainAlias during model loading (to convert to kOnnxDomain) or when creating messages (for clarity about what the domain was).

Reply:
https://github.com/microsoft/onnxruntime/pull/10962